resources/measures/upgrade_df_lighting_control/measure.rb

```diff
@@ -0,0 +1,425 @@
+    num_timesteps_in_hr.setDefaultValue(4)
+    args << num_timesteps_in_hr
+
+    choices = ['full baseline', 'bin sample', 'part year bin sample', 'fix', 'oat']
+    load_prediction_method = OpenStudio::Ruleset::OSArgument.makeChoiceArgument('load_prediction_method', choices, true)
+    load_prediction_method.setDisplayName("Method of load prediction (full baseline run, bin sample, part year bin sample, fixed schedule, outdoor air temperature-based)")
+    load_prediction_method.setDefaultValue('full baseline')
+    args << load_prediction_method
+
+    peak_lag = OpenStudio::Measure::OSArgument.makeIntegerArgument('peak_lag', true)
+    peak_lag.setDisplayName("Time lag of peak responding to temperature peak (hour), for oat prediction method only")
+    peak_lag.setDefaultValue(2)
+    args << peak_lag
+
+    choices_strate = ['max savings', 'start with peak', 'end with peak', 'center with peak']
+    peak_window_strategy = OpenStudio::Ruleset::OSArgument.makeChoiceArgument('peak_window_strategy', choices_strate, true)
+    peak_window_strategy.setDisplayName("Method of determining peak windows (max savings, start with peak, end with peak, center with peak)")
+    peak_window_strategy.setDefaultValue('center with peak')
+    args << peak_window_strategy
+
+    # apply_measure = OpenStudio::Measure::OSArgument.makeBoolArgument('apply_measure', true)
+    # apply_measure.setDisplayName('Apply measure?')
+    # apply_measure.setDescription('')
+    # apply_measure.setDefaultValue(true)
+    # args << apply_measure
+
+    return args
+  end
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)  # Do **NOT** remove this line
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+
+    ############################################
+    # assign the user inputs to variables
+    ############################################
+    peak_len = runner.getIntegerArgumentValue("peak_len",user_arguments)
+    light_adjustment_method = runner.getStringArgumentValue("light_adjustment_method",user_arguments)
+    light_adjustment = runner.getDoubleArgumentValue('light_adjustment', user_arguments)
+    num_timesteps_in_hr = runner.getIntegerArgumentValue("num_timesteps_in_hr",user_arguments)
+    load_prediction_method = runner.getStringArgumentValue("load_prediction_method",user_arguments)
+    peak_lag = runner.getIntegerArgumentValue("peak_lag",user_arguments)
+    peak_window_strategy = runner.getStringArgumentValue("peak_window_strategy",user_arguments)
+    # apply_measure = runner.getBoolArgumentValue('apply_measure', user_arguments)
+
+    def light_adj_based_on_sch(peak_sch, light_adjustment)
+      light_adj_values = peak_sch.map{|a| 1-light_adjustment/100.0*a}
+      return light_adj_values
+    end
+    
+    def get_hourly_schedule_from_schedule_ruleset(model, schedule_ruleset)
```

Comment:
Look at using this OpenstudioStandards::Schedule method: https://github.com/NREL/openstudio-standards/blob/8f948207d6af73165a2a8232559804b93d8e50c2/lib/openstudio-standards/schedules/information.rb#L106

Reply:
That method will only output hourly (8760) values. To get timestep values, you could replace almost everything in the `day_sched_array.each` block with this new (in 3.8.0) method [`ScheduleDay::timeSeries`](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.8.0-doc/model/html/classopenstudio_1_1model_1_1_schedule_day.html#a063fc4b5d4b6cc6979ce732e9bc314c7).